tests/cmdline/params/types/test_group.py

```diff
@@ -7,81 +7,140 @@
+    breaker character will force the identifier to be treated as a LABEL.
+    """
+    entity_01, entity_02, entity_03 = setup_groups
+    param = GroupParamType()
+    identifier = '{}'.format(entity_02.label)
+    result = param.convert(identifier, None, None)
+    assert result.uuid == entity_01.uuid
+
+    identifier = '{}{}'.format(entity_02.label, OrmEntityLoader.label_ambiguity_breaker)
+    result = param.convert(identifier, None, None)
+    assert result.uuid == entity_02.uuid
+
+
+def test_ambiguous_label_uuid(setup_groups):
+    """Situation: LABEL of entity_03 is exactly equal to UUID of entity_01.
+
+    Verify that using an ambiguous identifier gives precedence to the UUID interpretation. Appending the special
+    ambiguity breaker character will force the identifier to be treated as a LABEL.
+    """
+    entity_01, entity_02, entity_03 = setup_groups
+    param = GroupParamType()
+    identifier = '{}'.format(entity_03.label)
+    result = param.convert(identifier, None, None)
+    assert result.uuid == entity_01.uuid
+
+    identifier = '{}{}'.format(entity_03.label, OrmEntityLoader.label_ambiguity_breaker)
+    result = param.convert(identifier, None, None)
+    assert result.uuid == entity_03.uuid
+
+
+def test_create_if_not_exist(setup_groups):
+    """Test the `create_if_not_exist` constructor argument."""
+    label = 'non-existing-label-01'
+    param = GroupParamType(create_if_not_exist=True)
+    result = param.convert(label, None, None)
+    assert isinstance(result, Group)
+
+    label = 'non-existing-label-02'
+    param = GroupParamType(create_if_not_exist=True, sub_classes=('aiida.groups:core.auto',))
+    result = param.convert(label, None, None)
+    assert isinstance(result, AutoGroup)
+
+    # Specifying more than one subclass when `create_if_not_exist=True` is not allowed.
+    with pytest.raises(ValueError):
+        GroupParamType(create_if_not_exist=True, sub_classes=('aiida.groups:core.auto', 'aiida.groups:core.import'))
+
+
+@pytest.mark.parametrize(('sub_classes', 'expected'), (
+    (None, (True, True, True)),
+    (('aiida.groups:core.auto',), (False, True, False)),
+    (('aiida.groups:core.auto', 'aiida.groups:core.import'), (False, True, True)),
+))
+def test_sub_classes(setup_groups, sub_classes, expected):
+    """Test the `sub_classes` constructor argument."""
+    entity_01, entity_02, entity_03 = setup_groups
+    param = GroupParamType(sub_classes=sub_classes)
+
+    results = []
+
+    for group in [entity_01, entity_02, entity_03]:
+        try:
+            param.convert(str(group.pk), None, None)
+        except click.BadParameter:
+            results.append(False)
+        else:
+            results.append(True)
+
+    assert tuple(results) == expected
+
+
+def test_complete(setup_groups):
+    """Test the `complete` method that provides auto-complete functionality."""
+    entity_01, entity_02, entity_03 = setup_groups
+    entity_04 = Group(label='xavier').store()
```

Comment:
Been catching up on some X-Men or? 😅

Reply:
Not really, just needed something with a letter late in the alphabet. Apparently bald psychic men are lurking in my subconscious :smile: